scripts/render_DESeq2_report.R

```diff
@@ -105,4 +110,15 @@ if (is.na(params$group_facet)) {
                       params = params,
                       envir = new.env())
   }
+  deg_files <- fs::dir_ls(file.path(config$DESeq2$projectdir, "DEG_output"),
+                          regexp = "\\-DEG_summary.txt$", recurse = T)
+  # This depends on 'cat' being available on the command line (i.e., linux-specific)
+  # Also some insane quoting going on here, but I don't see an easier way
+  system(paste0('cat "', paste(deg_files, collapse='"  "'),
+                '"  >  ', file.path(config$DESeq2$projectdir,
+                                 "DEG_output/DEG_summary.txt")))
+  
+  # This would probably fail in cases where different numbers of contrasts exists across facets.
+  # But could otherwise be useful?
+  # results <- deg_files %>% map_dfr(read_tsv, col_names=T, .id="source") 
```

Comment:
Is there a reason you're doing this on the command line instead of reading them into R?

Reply:
Yes, actually. I initially wanted to do it in R.

The reason I'm doing in bash is that we often have different numbers of columns in these tables. Some chemicals might have entire doses drop out. Some, by design, may have different numbers of concentrations/whatever other variables. So simply concatenating everything eliminates any possibility that R will complain about tables with different dimensions.

For instance, with the PFAS project, we have - I kid you not - 54 combinations of timepoint and chemical. So, having these simple text outputs lets the end user easily take all that information to make their own summary table. It's not pretty, but it saves them the trouble of opening 54 HTML files and copy/pasting the DEG summary tables from each one.

However, you're completely right, you could read them into R, take the longest width of all of them, then change the dimensions of each to add empty space, and rbind them together... There might be an easier way? If there is, please let me know. But otherwise it seems like a lot of extra steps to do it in R. Not opposed to it at all, but I don't know offhand how to readily deal with the different dimensions. Bash just seemed way simpler.

Reply:
I think you could just read the whole thing into a single character vector and then write it out (essentially the same as what cat does but avoiding the system call) but I'm not super fussed about it. I'm not planning on running this on windows anyway.